src/Employer/Employer.Web/Services/EmployerAccountTypeCookieWriter.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Security.Cryptography;
+using Esfa.Recruit.Employer.Web.Configuration;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Esfa.Recruit.Employer.Web.Services
+{
+    public class EmployerAccountTypeCookieWriter : IEmployerAccountTypeCookieWriter
+    {
+        private readonly IHostingEnvironment _hostingEnvironment;
+        private readonly IDataProtector _dataProtector;
+        private readonly ILogger<EmployerAccountTypeCookieWriter> _logger;
+
+        public EmployerAccountTypeCookieWriter(
+            IHostingEnvironment hostingEnvironment,
+            IDataProtectionProvider dataProtectionProvider,
+            ILogger<EmployerAccountTypeCookieWriter> logger)
+        {
+            _hostingEnvironment = hostingEnvironment;
+            _dataProtector = dataProtectionProvider.CreateProtector(DataProtectionPurposes.EmployerAccountTypeCookie);
```

Comment:
Do you think we should maybe declare this as a singleton or passed in from the Ioc for creating `EmployerAccountTypeCookieWriter`?

Reply:
I don't understand. This class is only used via dependency injection by its interface.